pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -192,6 +192,7 @@ public void findEntryFailed(ManagedLedgerException exception, Optional<Position>
             log.warn("[{}][{}] read failed from ledger at position:{} : {}", topicName, subName, failedReadPosition,
                     exception.getMessage());
             findEntryComplete(failedReadPosition.get(), ctx);
```

Comment:
It seems that here we just want to handle `findEntryFailed` like `findEntryComplete`  if  `autoSkipNonRecoverableData=true`.   And `findEntryComplete` will change `expirationCheckInProgress` to false.  So this change looks reasonable I think @Pomelongan @codelipenghui

Reply:
@AnonHxy If we have completed the find entry task either succeeded or failed, we change `expirationCheckInProgress` to false, no? Otherwise, it looks like we will set `expirationCheckInProgress=true` before starting to find the entry, but the findEntryComplete and findEntryFailed can't ensure the `expirationCheckInProgress` change to false finally, it sounds confused.

Reply:
@codelipenghui `expirationCheckInProgress` will be set to false in this `findEntryComplete(failedReadPosition.get(), ctx);`.